accounts-db/src/account_locks.rs

```diff
@@ -110,6 +114,48 @@ impl AccountLocks {
     }
 }
 
+/// Validate account locks before locking.
+pub fn validate_account_locks(
```

Comment:
now we have duplicated implementations from SDK, not sure what is the best way to keep them synced in the future (in case there's more `else if` to add, for example). Maybe add comments on both places?

Reply:
Now that you mention it - I should just deprecate the SDK one.

I moved it there at some point from `bank`.
No reason it needs to be there any more, and I actually would like to guarantee we're not calling it elsewhere!

Reply:
ah nvm I just seperated the validate fn from `get_account_locks`.

I think with the recent changes we don't actually use the `get_account_locks` anymore (nor should we as it allocates). May need to deprecate that.

Reply:
@tao-stones

There are several uses of the `get_account_locks` and `validate_account_locks` from `SanitizedTransaction`.

With the code in this PR I can remove / transition those uses to this new code - but there are several uses across different modules, many requiring some changes to fn args.

wdyt of this plan?
1. merge this PR with duplciate logic (after I address your comments)
2. create PRs to use new code where needed
3. deprecate the `get_account_locks` and related fns (unsure if this has downstream effects)

Reply:
sounds good, maybe can deprecate sdk version sooner